src/activities/quizzes/ipRestrictions/QuizIpRestrictionsEntity.js

```diff
@@ -0,0 +1,101 @@
+import { Entity } from '../../../es6/Entity';
+import { Actions, Classes } from '../../../hypermedia-constants';
+import { performSirenAction } from '../../../es6/SirenAction.js';
+
+const IP_START_FIELD = 'start';
+const IP_END_FIELD = 'end';
+
+/**
+ * Quiz IP restrictions subentity of a d2l Quiz.
+ */
+export class QuizIpRestrictionsEntity extends Entity {
+
+	/**
+	 * @returns {bool} Whether or not the edit IP restrictions type action is present
+	 */
+
+	canEditIpRestrictions() {
+		return this._entity && this._entity.hasActionByName(Actions.quizzes.ipRestrictions.add);
+	}
+
+	/**
+	 * @returns {object} Returns IP restriction sub entity
+	 */
+
+	getIpRestrictions() {
+		if (!this._entity) {
+			return;
+		}
+
+		const ipRestrictionEntities = this._entity.getSubEntitiesByClass(Classes.quizzes.ip.range);
+
+		if (!ipRestrictionEntities || !ipRestrictionEntities.length) {
```

Comment:
If there are no current IP restrictions then I believe we want the table to render a single empty row which the user can fill in. Not sure if this is the best place to make that decision.